src/sentry/incidents/metric_issue_detector.py

```diff
@@ -241,7 +241,9 @@ def is_editing_transaction_dataset(
             return True
         return False
 
-    def update_data_source(self, instance: Detector, data_source: SnubaQueryDataSourceType):
+    def update_data_source(
+        self, instance: Detector, data_source: SnubaQueryDataSourceType, seer_updated: bool = False
```

Comment:
I had to introduce `seer_updated` to avoid calling Seer twice in the event that the type _and_ snuba query are changing

Reply:
The test I added does cover this